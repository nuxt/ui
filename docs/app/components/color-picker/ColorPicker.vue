<template>
  <UPopover mode="hover" :ui="{ content: 'p-2' }">
    <template #default="{ open }">
      <UButton
        icon="i-heroicons-swatch-20-solid"
        color="neutral"
        :variant="open ? 'soft' : 'ghost'"
        square
        aria-label="Color picker"
        :ui="{ leadingIcon: 'text-[--ui-primary]' }"
      />
    </template>

    <template #content>
      <fieldset class="grid grid-cols-5 gap-px">
        <legend class="text-[11px] font-bold mb-1">
          Primary
        </legend>

        <ColorPickerPill v-for="color in primaryColors" :key="color" :color="color" :selected="primary" @select="primary = color" />
      </fieldset>

      <USeparator class="my-2" type="dashed" />

      <fieldset class="grid grid-cols-5 gap-px">
        <legend class="text-[11px] font-bold mb-1">
          Neutral
        </legend>

        <ColorPickerPill v-for="color in neutralColors" :key="color" :color="color" :selected="neutral" @select="neutral = color" />
      </fieldset>
    </template>
  </UPopover>
</template>

<script setup lang="ts">
import colors from 'tailwindcss/colors'
import { omit } from '#ui/utils'

const appConfig = useAppConfig()

// Computed

const neutralColors = ['slate', 'gray', 'zinc', 'neutral', 'stone']
const neutral = computed({
  get() {
    return appConfig.ui.colors.neutral
  },
  set(option) {
    appConfig.ui.colors.neutral = option
    window.localStorage.setItem('nuxt-ui-neutral', appConfig.ui.colors.neutral)
  }
})

const colorsToOmit = ['inherit', 'current', 'transparent', 'black', 'white', ...neutralColors]
const primaryColors = Object.keys(omit(colors, colorsToOmit as any))
const primary = computed({
  get() {
    return appConfig.ui.colors.primary
  },
  set(option) {
    appConfig.ui.colors.primary = option
    window.localStorage.setItem('nuxt-ui-primary', appConfig.ui.colors.primary)
  }
})
</script>
