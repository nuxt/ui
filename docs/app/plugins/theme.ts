export default defineNuxtPlugin({
  enforce: 'post',
  setup() {
    const appConfig = useAppConfig()

    if (import.meta.client) {
      function updateColor(type: 'primary' | 'neutral') {
        const color = localStorage.getItem(`nuxt-ui-${type}`)
        if (color) {
          appConfig.ui.colors[type] = color
        }
      }

      function updateRadius() {
        const radius = localStorage.getItem('nuxt-ui-radius')
        if (radius) {
          appConfig.theme.radius = Number.parseFloat(radius)
        }
      }

      updateColor('primary')
      updateColor('neutral')
      updateRadius()
    }

    if (import.meta.server) {
      useHead({
        script: [{
          innerHTML: `
            let html = document.querySelector('style#nuxt-ui-colors').innerHTML;

            if (localStorage.getItem('nuxt-ui-primary')) {
              const primaryColor = localStorage.getItem('nuxt-ui-primary');
              html = html.replace(
                /(--ui-color-primary-\\d{2,3}:\\s*var\\()--color-${appConfig.ui.colors.primary}-(\\d{2,3}\\))/g,
                \`$1--color-\${primaryColor}-$2\`
              );
            }
            if (localStorage.getItem('nuxt-ui-neutral')) {
              const neutralColor = localStorage.getItem('nuxt-ui-neutral');
              html = html.replace(
                /(--ui-color-neutral-\\d{2,3}:\\s*var\\()--color-${appConfig.ui.colors.neutral}-(\\d{2,3}\\))/g,
                \`$1--color-\${neutralColor}-$2\`
              );
            }

            document.querySelector('style#nuxt-ui-colors').innerHTML = html;
            `.replace(/\s+/g, ' '),
          type: 'text/javascript',
          tagPriority: -1
        }, {
          innerHTML: `
            if (localStorage.getItem('nuxt-ui-radius')) {
              document.querySelector('style#nuxt-ui-radius').innerHTML = ':root { --ui-radius: ' + localStorage.getItem('nuxt-ui-radius') + 'rem; }';
            }
          `.replace(/\s+/g, ' '),
          type: 'text/javascript',
          tagPriority: -1
        }]
      })
    }
  }
})
