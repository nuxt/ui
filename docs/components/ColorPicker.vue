<template>
  <div class="p-2">
    <div class="grid grid-cols-5 gap-px">
      <ColorPickerButton v-for="color in primaryColors" :key="color.value" :color="color" :selected="primary" @select="primary = color" />
    </div>

    <hr class="border-gray-200 dark:border-gray-800 my-2">

    <div class="grid grid-cols-5 gap-px">
      <ColorPickerButton v-for="color in grayColors" :key="color.value" :color="color" :selected="gray" @select="gray = color" />
    </div>
  </div>
</template>

<script setup lang="ts">
import colors from '#tailwind-config/theme/colors'

const appConfig = useAppConfig()
const colorMode = useColorMode()

// Computed

const primaryColors = computed(() => useWithout(appConfig.ui.colors, 'primary').map(color => ({ value: color, text: color, hex: colors[color][colorMode.value === 'dark' ? 400 : 500] })))
const primary = computed({
  get () {
    return primaryColors.value.find(option => option.value === appConfig.ui.primary)
  },
  set (option) {
    appConfig.ui.primary = option.value

    window.localStorage.setItem('nuxt-ui-primary', appConfig.ui.primary)
  }
})

const grayColors = computed(() => ['slate', 'cool', 'zinc', 'neutral', 'stone'].map(color => ({ value: color, text: color, hex: colors[color][colorMode.value === 'dark' ? 400 : 500] })))
const gray = computed({
  get () {
    return grayColors.value.find(option => option.value === appConfig.ui.gray)
  },
  set (option) {
    appConfig.ui.gray = option.value

    window.localStorage.setItem('nuxt-ui-gray', appConfig.ui.gray)
  }
})
</script>
