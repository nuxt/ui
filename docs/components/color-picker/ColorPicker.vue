<template>
  <UPopover mode="hover">
    <template #default="{ open }">
      <UButton color="gray" variant="ghost" square :class="[open && 'bg-gray-50 dark:bg-gray-800']" aria-label="Color picker">
        <UIcon name="i-heroicons-swatch-20-solid" class="w-5 h-5 text-primary-500 dark:text-primary-400" />
      </UButton>
    </template>

    <template #panel>
      <div class="p-2">
        <div class="grid grid-cols-5 gap-px">
          <ColorPickerPill v-for="color in primaryColors" :key="color.value" :color="color" :selected="primary" @select="primary = color" />
        </div>

        <hr class="border-gray-200 dark:border-gray-800 my-2">

        <div class="grid grid-cols-5 gap-px">
          <ColorPickerPill v-for="color in grayColors" :key="color.value" :color="color" :selected="gray" @select="gray = color" />
        </div>
      </div>
    </template>
  </UPopover>
</template>

<script setup lang="ts">
import colors from '#tailwind-config/theme/colors'

const appConfig = useAppConfig()
const colorMode = useColorMode()

// Computed

const primaryColors = computed(() => appConfig.ui.colors.filter(color => color !== 'primary').map(color => ({ value: color, text: color, hex: colors[color][colorMode.value === 'dark' ? 400 : 500] })))
const primary = computed({
  get () {
    return primaryColors.value.find(option => option.value === appConfig.ui.primary)
  },
  set (option) {
    appConfig.ui.primary = option.value

    window.localStorage.setItem('nuxt-ui-primary', appConfig.ui.primary)
  }
})

const grayColors = computed(() => ['slate', 'cool', 'zinc', 'neutral', 'stone'].map(color => ({ value: color, text: color, hex: colors[color][colorMode.value === 'dark' ? 400 : 500] })))
const gray = computed({
  get () {
    return grayColors.value.find(option => option.value === appConfig.ui.gray)
  },
  set (option) {
    appConfig.ui.gray = option.value

    window.localStorage.setItem('nuxt-ui-gray', appConfig.ui.gray)
  }
})
</script>
