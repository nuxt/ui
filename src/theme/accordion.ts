export default {
  slots: {
    root: 'w-full',
    item: 'border-b border-[var(--ui-border)] last:border-b-0',
    header: 'flex',
    trigger: 'group flex-1 flex items-center gap-1.5 font-medium text-sm py-3.5 focus-visible:outline-[var(--ui-primary)] min-w-0',
    content: 'data-[state=open]:animate-[accordion-down_200ms_ease-out] data-[state=closed]:animate-[accordion-up_200ms_ease-out] overflow-hidden focus:outline-none',
    body: 'text-sm pb-3.5',
    leadingIcon: 'shrink-0 size-5',
    trailingIcon: 'shrink-0 size-5 ms-auto group-data-[state=open]:rotate-180 transition-transform duration-200',
    label: 'text-start break-words'
  },
  variants: {
    disabled: {
      true: {
        trigger: 'cursor-not-allowed opacity-75'
      }
    }
  }
}
