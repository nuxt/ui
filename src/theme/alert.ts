import type { ModuleOptions } from '../module'

export default (options: Required<ModuleOptions>) => ({
  slots: {
    root: 'relative overflow-hidden w-full rounded-[calc(var(--ui-radius)*2)] p-4 flex gap-2.5',
    wrapper: 'min-w-0 flex-1 flex flex-col gap-1',
    title: 'text-sm font-medium',
    description: 'text-sm opacity-90',
    icon: 'shrink-0 size-5',
    avatar: 'shrink-0',
    avatarSize: '2xl',
    actions: 'flex gap-1.5 shrink-0',
    close: 'p-0.5'
  },
  variants: {
    color: {
      ...Object.fromEntries((options.theme.colors || []).map((color: string) => [color, ''])),
      neutral: ''
    },
    variant: {
      solid: '',
      outline: '',
      soft: '',
      subtle: ''
    },
    multiline: {
      true: {
        root: 'items-start',
        actions: 'items-start mt-1'
      },
      false: {
        root: 'items-center',
        actions: 'items-center'
      }
    }
  },
  compoundVariants: [...(options.theme.colors || []).map((color: string) => ({
    color,
    variant: 'solid',
    class: {
      root: `bg-[var(--ui-${color})] text-[var(--ui-bg)]`
    }
  })), ...(options.theme.colors || []).map((color: string) => ({
    color,
    variant: 'outline',
    class: {
      root: `text-[var(--ui-${color})] ring ring-inset ring-[var(--ui-${color})]/25`
    }
  })), ...(options.theme.colors || []).map((color: string) => ({
    color,
    variant: 'soft',
    class: {
      root: `bg-[var(--ui-${color})]/10 text-[var(--ui-${color})]`
    }
  })), ...(options.theme.colors || []).map((color: string) => ({
    color,
    variant: 'subtle',
    class: {
      root: `bg-[var(--ui-${color})]/10 text-[var(--ui-${color})] ring ring-inset ring-[var(--ui-${color})]/25`
    }
  })), {
    color: 'neutral',
    variant: 'solid',
    class: {
      root: 'text-[var(--ui-bg)] bg-[var(--ui-bg-inverted)]'
    }
  }, {
    color: 'neutral',
    variant: 'outline',
    class: {
      root: 'text-[var(--ui-text-highlighted)] bg-[var(--ui-bg)] ring ring-inset ring-[var(--ui-border)]'
    }
  }, {
    color: 'neutral',
    variant: 'soft',
    class: {
      root: 'text-[var(--ui-text-highlighted)] bg-[var(--ui-bg-elevated)]/50'
    }
  }, {
    color: 'neutral',
    variant: 'subtle',
    class: {
      root: 'text-[var(--ui-text-highlighted)] bg-[var(--ui-bg-elevated)]/50 ring ring-inset ring-[var(--ui-border-accented)]'
    }
  }],
  defaultVariants: {
    color: 'primary',
    variant: 'solid'
  }
})
