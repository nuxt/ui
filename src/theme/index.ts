export { default as accordion } from './accordion'
export { default as alert } from './alert'
export { default as avatar } from './avatar'
export { default as avatarGroup } from './avatar-group'
export { default as badge } from './badge'
export { default as breadcrumb } from './breadcrumb'
export { default as button } from './button'
export { default as buttonGroup } from './button-group'
export { default as calendar } from './calendar'
export { default as card } from './card'
export { default as carousel } from './carousel'
export { default as checkbox } from './checkbox'
export { default as chip } from './chip'
export { default as collapsible } from './collapsible'
export { default as colorPicker } from './color-picker'
export { default as commandPalette } from './command-palette'
export { default as container } from './container'
export { default as contextMenu } from './context-menu'
export { default as drawer } from './drawer'
export { default as dropdownMenu } from './dropdown-menu'
export { default as form } from './form'
export { default as formField } from './form-field'
export { default as input } from './input'
export { default as inputMenu } from './input-menu'
export { default as inputNumber } from './input-number'
export { default as kbd } from './kbd'
export { default as link } from './link'
export { default as modal } from './modal'
export { default as navigationMenu } from './navigation-menu'
export { default as pagination } from './pagination'
export { default as pinInput } from './pin-input'
export { default as popover } from './popover'
export { default as progress } from './progress'
export { default as radioGroup } from './radio-group'
export { default as select } from './select'
export { default as selectMenu } from './select-menu'
export { default as separator } from './separator'
export { default as skeleton } from './skeleton'
export { default as slideover } from './slideover'
export { default as slider } from './slider'
export { default as stepper } from './stepper'
export { default as switch } from './switch'
export { default as table } from './table'
export { default as tabs } from './tabs'
export { default as textarea } from './textarea'
export { default as toast } from './toast'
export { default as toaster } from './toaster'
export { default as tooltip } from './tooltip'
