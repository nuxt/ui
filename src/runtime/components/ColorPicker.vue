<script lang="ts">
import { tv, type VariantProps } from 'tailwind-variants'
import type { MaybeRefOrGetter } from '@vueuse/shared'
import type { AppConfig } from '@nuxt/schema'
import _appConfig from '#build/app.config'
import theme from '#build/ui/color-picker'

const appConfig = _appConfig as AppConfig & { ui: { colorPicker: Partial<typeof theme> } }

const colorPicker = tv({ extend: tv(theme), ...(appConfig.ui?.colorPicker || {}) })

type ColorPickerVariants = VariantProps<typeof colorPicker>

type HSVColor = {
  h: number
  s: number
  v: number
}

export type ColorPickerProps = {
  /**
   * The element or component this component should render as.
   * @defaultValue 'div'
   */
  as?: any
  /**
   * Throttle time in ms for the color picker
   */
  throttle?: number
  /**
   * Disable the color picker
   */
  disabled?: boolean
  /**
   * The default value of the color picker
   */
  defaultValue?: string
  /**
   * Format of the color
   * @defaultValue 'hex'
   */
  format?: 'hex' | 'rgb' | 'hsl' | 'hwb'
  size?: ColorPickerVariants['size']
  class?: any
  ui?: Partial<typeof colorPicker.slots>
}
</script>

<script setup lang="ts">
import { ref, nextTick, computed, toValue } from 'vue'
import { Primitive } from 'reka-ui'
import { useEventListener, useElementBounding, watchThrottled, watchPausable } from '@vueuse/core'
import { isClient } from '@vueuse/shared'
import Color from 'color'

const props = withDefaults(defineProps<ColorPickerProps>(), {
  format: 'hex',
  throttle: 50,
  defaultValue: '#FFFFFF'
})
const modelValue = defineModel<string>(undefined)

const pickedColor = computed<HSVColor>({
  get() {
    try {
      const color = Color(modelValue.value || props.defaultValue)
      return color.hsv().object() as HSVColor
    } catch (_) {
      return { h: 0, s: 0, v: 100 }
    }
  },
  set(value) {
    const color = Color.hsv(value.h, value.s, value.v)

    switch (props.format) {
      case 'rgb':
        modelValue.value = color.rgb().string()
        break
      case 'hsl':
        modelValue.value = color.hsl().string()
        break
      case 'hwb':
        modelValue.value = color.hwb().string()
        break
      case 'hex':
      default:
        modelValue.value = color.hex()
    }
  }
})

function useColorDraggable(targetElement: MaybeRefOrGetter<HTMLElement | null>,
  containerElement: MaybeRefOrGetter<HTMLElement | null>,
  axis: 'x' | 'y' | 'both' = 'both',
  initialPosition = { x: 0, y: 0 },
  disabled?: MaybeRefOrGetter<boolean | undefined>
) {
  const position = ref<{ x: number, y: number }>(initialPosition)
  const pressedDelta = ref<{ x: number, y: number }>()
  const targetRect = useElementBounding(targetElement)
  const containerRect = useElementBounding(containerElement)

  function start(event: PointerEvent) {
    if (toValue(disabled)) return event.preventDefault()

    const container = toValue(containerElement)

    pressedDelta.value = {
      x: event.clientX - (container ? event.clientX - containerRect.left.value + container.scrollLeft : targetRect.left.value),
      y: event.clientY - (container ? event.clientY - containerRect.top.value + container.scrollTop : targetRect.top.value)
    }

    move(event)
  }

  function move(event: PointerEvent) {
    if (!pressedDelta.value) return

    const container = toValue(containerElement)
    let { x, y } = position.value

    if (container && (axis === 'x' || axis === 'both')) {
      x = Math.min(Math.max(0, (event.clientX - pressedDelta.value.x) / container.scrollWidth * 100), 100)
    }

    if (container && (axis === 'y' || axis === 'both')) {
      y = Math.min(Math.max(0, (event.clientY - pressedDelta.value.y) / container.scrollHeight * 100), 100)
    }

    position.value = { x, y }
  }

  function end() {
    if (!pressedDelta.value) {
      return
    }

    pressedDelta.value = undefined
  }

  if (isClient) {
    useEventListener(containerElement, 'pointerdown', start)
    useEventListener(window, 'pointermove', move)
    useEventListener(window, 'pointerup', end)
  }

  return {
    position
  }
}

function normalizeHue(hue: number, dir: 'left' | 'right' = 'left'): number {
  if (dir === 'right') {
    return (hue * 100) / 360
  }

  return (hue / 100) * 360
}

function normalizeBrightness(brightness: number): number {
  return 100 - brightness
}

const selectorRef = ref<HTMLDivElement | null>(null)
const selectorThumbRef = ref<HTMLDivElement | null>(null)
const trackRef = ref<HTMLDivElement | null>(null)
const trackThumbRef = ref<HTMLDivElement | null>(null)

const disabled = computed(() => props.disabled)

const { position: selectorThumbPosition } = useColorDraggable(selectorThumbRef, selectorRef, 'both', {
  x: pickedColor.value.s,
  y: normalizeBrightness(pickedColor.value.v)
}, disabled)

const { position: trackThumbPosition } = useColorDraggable(trackThumbRef, trackRef, 'y', {
  x: 0,
  y: normalizeHue(pickedColor.value.h, 'right')
}, disabled)

const { pause: pauseWatchColor, resume: resumeWatchColor } = watchPausable(pickedColor, (hsb) => {
  selectorThumbPosition.value = {
    x: hsb.s,
    y: normalizeBrightness(hsb.v)
  }
  trackThumbPosition.value = {
    x: 0,
    y: normalizeHue(hsb.h, 'right')
  }
})

watchThrottled([selectorThumbPosition, trackThumbPosition], () => {
  pauseWatchColor()

  pickedColor.value = {
    h: normalizeHue(trackThumbPosition.value.y),
    s: selectorThumbPosition.value.x,
    v: normalizeBrightness(selectorThumbPosition.value.y)
  }

  nextTick(resumeWatchColor)
}, { throttle: () => props.throttle })

const trackThumbColor = computed(() => Color({
  h: normalizeHue(trackThumbPosition.value.y),
  s: 100,
  v: 100
}).hex())

const selectorStyle = computed(() => ({
  backgroundColor: trackThumbColor.value
}))

const selectorThumbStyle = computed(() => ({
  backgroundColor: Color(modelValue.value || props.defaultValue).hex(),
  left: `${selectorThumbPosition.value.x}%`,
  top: `${selectorThumbPosition.value.y}%`
}))

const trackThumbStyle = computed(() => ({
  backgroundColor: trackThumbColor.value,
  top: `${trackThumbPosition.value.y}%`
}))

const ui = computed(() => colorPicker({
  size: props.size
}))
</script>

<template>
  <Primitive :as="as" :class="ui.root({ class: [props.class, props.ui?.root] })" :data-disabled="disabled ? true : undefined">
    <div :class="ui.picker({ class: props.ui?.picker })">
      <div
        ref="selectorRef"
        :class="ui.selector({ class: props.ui?.selector })"
        :style="selectorStyle"
      >
        <div :class="ui.selectorBackground({ class: props.ui?.selectorBackground })" data-color-picker-background>
          <div
            ref="selectorThumbRef"
            :class="ui.selectorThumb({ class: props.ui?.selectorThumb })"
            :style="selectorThumbStyle"
            :data-disabled="disabled ? true : undefined"
          />
        </div>
      </div>
      <div
        ref="trackRef"
        :class="ui.track({ class: props.ui?.track })"
        data-color-picker-track
      >
        <div
          ref="trackThumbRef"
          :class="ui.trackThumb({ class: props.ui?.trackThumb })"
          :style="trackThumbStyle"
          :data-disabled="disabled ? true : undefined"
        />
      </div>
    </div>
  </Primitive>
</template>

<style scoped>
[data-color-picker-background] {
  background-image: linear-gradient(to top, #000 0%, rgba(0, 0, 0, 0) 100%), linear-gradient(to right, #fff 0%, rgba(255, 255, 255, 0) 100%);
}

[data-color-picker-track] {
  background-image: linear-gradient(0deg, red 0, #f0f 17%, #00f 33%, #0ff 50%, #0f0 67%, #ff0 83%, red);
}
</style>
