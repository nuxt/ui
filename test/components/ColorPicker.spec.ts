import { describe, it, expect } from 'vitest'
import ColorPicker, { type ColorPickerProps, type ColorPickerSlots } from '../../src/runtime/components/ColorPicker.vue'
import ComponentRender from '../component-render'
import theme from '#build/ui/color-picker'

describe('ColorPicker', () => {
  const sizes = Object.keys(theme.variants.size) as any

  it.each([
    // Props
    ['with inline', { props: { inline: true } }],
    ['with disabled', { props: { disabled: true } }],
    ['with as', { props: { as: 'div' } }],
    ['with class', { props: { class: 'w-96' } }],
    ['with ui', { props: { ui: { root: 'w-96' } } }],
    ...sizes.map((size: string) => [`with size ${size}`]),
    // Slots
    ['with trigger slot', { slots: { trigger: () => 'Trigger' } }]
  ])('renders %s correctly', async (nameOrHtml: string, options: { props?: ColorPickerProps, slots?: Partial<ColorPickerSlots> }) => {
    const html = await ComponentRender(nameOrHtml, options, ColorPicker)
    expect(html).toMatchSnapshot()
  })
})
