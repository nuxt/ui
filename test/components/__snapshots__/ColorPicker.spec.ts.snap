// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ColorPicker > renders with as correctly 1`] = `
"<div class=""><button class="rounded-[calc(var(--ui-radius)*1.5)] inline-flex p-1.5 ring ring-inset ring-[var(--ui-border-accented)] disabled:cursor-not-allowed focus:outline-none aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-[var(--ui-primary)] hover:bg-[var(--ui-bg-elevated)] transition-colors" type="button" aria-haspopup="dialog" aria-expanded="false" aria-controls="" data-state="closed"><span class="rounded-[var(--ui-radius)] w-5 h-5" style="background-color: #f00000;"></span></button>
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;

exports[`ColorPicker > renders with class correctly 1`] = `
"<div class="w-96"><button class="rounded-[calc(var(--ui-radius)*1.5)] inline-flex p-1.5 ring ring-inset ring-[var(--ui-border-accented)] disabled:cursor-not-allowed focus:outline-none aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-[var(--ui-primary)] hover:bg-[var(--ui-bg-elevated)] transition-colors" type="button" aria-haspopup="dialog" aria-expanded="false" aria-controls="" data-state="closed"><span class="rounded-[var(--ui-radius)] w-5 h-5" style="background-color: #f00000;"></span></button>
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;

exports[`ColorPicker > renders with disabled correctly 1`] = `
"<div class=""><button class="rounded-[calc(var(--ui-radius)*1.5)] inline-flex p-1.5 ring ring-inset ring-[var(--ui-border-accented)] disabled:cursor-not-allowed focus:outline-none aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-[var(--ui-primary)] hover:bg-[var(--ui-bg-elevated)] transition-colors" disabled="" type="button" aria-haspopup="dialog" aria-expanded="false" aria-controls="" data-state="closed"><span class="rounded-[var(--ui-radius)] w-5 h-5" style="background-color: #f00000;"></span></button>
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;

exports[`ColorPicker > renders with inline correctly 1`] = `
"<div class="">
  <div class="flex gap-4">
    <div class="rounded w-42 h-42" style="background-color: #f00000;">
      <div class="w-full h-full relative rounded" data-color-picker-background="">
        <div class="absolute size-4 ring-2 ring-[var(--color-white)] rounded-full cursor-pointer" style="background-color: #ffffff;"></div>
      </div>
    </div>
    <div class="w-[8px] rounded relative h-42" data-color-picker-track="">
      <div class="absolute transform -translate-x-[4px] size-4 rounded-full ring-2 ring-[var(--color-white)] cursor-pointer" style="background-color: #f00000;"></div>
    </div>
  </div>
</div>"
`;

exports[`ColorPicker > renders with size lg correctly 1`] = `"with size lg"`;

exports[`ColorPicker > renders with size md correctly 1`] = `"with size md"`;

exports[`ColorPicker > renders with size sm correctly 1`] = `"with size sm"`;

exports[`ColorPicker > renders with size xl correctly 1`] = `"with size xl"`;

exports[`ColorPicker > renders with size xs correctly 1`] = `"with size xs"`;

exports[`ColorPicker > renders with trigger slot correctly 1`] = `
"<div class="">Trigger
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;

exports[`ColorPicker > renders with ui correctly 1`] = `
"<div class="w-96"><button class="rounded-[calc(var(--ui-radius)*1.5)] inline-flex p-1.5 ring ring-inset ring-[var(--ui-border-accented)] disabled:cursor-not-allowed focus:outline-none aria-disabled:cursor-not-allowed disabled:opacity-75 aria-disabled:opacity-75 focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-[var(--ui-primary)] hover:bg-[var(--ui-bg-elevated)] transition-colors" type="button" aria-haspopup="dialog" aria-expanded="false" aria-controls="" data-state="closed"><span class="rounded-[var(--ui-radius)] w-5 h-5" style="background-color: #f00000;"></span></button>
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;
